bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitCommandExtension.java

```diff
@@ -68,25 +68,27 @@ public void execute(String[] args, Console console) {
                     break;
                 case SUBCMD_PRINT_ACCESSORY:
                     if (args.length > 1) {
-                        Integer id = Integer.valueOf(args[1]);
-                        printAccessory(id, console);
+                        printAccessory(args[1], console);
                     } else {
-                        console.println("accessory id is required as an argument");
+                        console.println("accessory id or name is required as an argument");
                     }
                     break;
                 default:
                     console.println("Unknown command '" + subCommand + "'");
                     printUsage(console);
                     break;
             }
+        } else {
+            printUsage(console);
         }
     }
 
     @Override
     public List<String> getUsages() {
         return Arrays.asList(buildCommandUsage(SUBCMD_LIST_ACCESSORIES, "list all HomeKit accessories"),
-                buildCommandUsage(SUBCMD_PRINT_ACCESSORY + " <accessory id>", "print accessorty details"),
-                buildCommandUsage(SUBCMD_CLEAR_PAIRINGS, "removes all pairings with HomeKit clients"),
+                buildCommandUsage(SUBCMD_PRINT_ACCESSORY + " <accessory id | accessory name>",
+                        "search for accessories and show details. "),
```

Comment:
Please mention the partial matching here too.

Reply:
good point. copied the description from README.